### PR TITLE
Adding "clean" Swagger validation test

### PR DESCRIPTION
**Pull request by tbombach**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@tbombach__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

File: src/modeler/AutoRest.Swagger.Tests/SwaggerModelerValidationTests.cs [hunk 1]
```diff
@@ -40,6 +40,16 @@ private IEnumerable<ValidationMessage> ValidateSwagger(string input)
...
         }
 
+        /// <summary>
+        /// Verifies that a clean Swagger file does not result in any validation errors
+        /// </summary>
+        [Fact]
+        public void CleanFileValidation()
+        {
+            var messages = ValidateSwagger(Path.Combine("Swagger", "Validation", "clean-complex-spec.json"));
+            Assert.Equal(0, messages.Where(m => m.Severity >= LogEntrySeverity.Warning).Count());
```
Comment (fearthecowboy) [inline src/modeler/AutoRest.Swagger.Tests/SwaggerModelerValidationTests.cs]: You might consider :

``` csharp
Assert.Empty(messages.Where(m => m.Severity >= LogEntrySeverity.Warning)); 
```

Which has the virtue, that if the collection takes a lot of time to count all the elements, assert will fail faster.

**Comment by fearthecowboy** [comment]
Just my one optimization; 

LGTM 👍

File: src/modeler/AutoRest.Swagger.Tests/SwaggerModelerValidationTests.cs (hunk 1 above)
Comment (tbombach) [inline src/modeler/AutoRest.Swagger.Tests/SwaggerModelerValidationTests.cs]: Good point!

**Pull request merged**